File: src/institution/interlink/Internship.js
var University = require('../University');
var Student = require('../../person/Student');

class Internship {
    constructor(name, averageKnowledge) {
        this.internshipName = name;
        this.internshipStudents = [];
        this.averageKnowledge = averageKnowledge;
    }

    setStudent(student) {
        if(student.getKnowledge() > this.averageKnowledge){
            this.internshipStudents.push(student);
        }
    }

    getStudents() {
        for (let intern of this.internshipStudents){
            console.log(`    ${intern.getName()} \t \t ${intern.getKnowledge()}`)
        }
    }
}
module.exports = Internship;

File: src/institution/interlink/Internship.java
package institution.interlink;

import java.util.ArrayList;
import java.util.List;

import institution.University;
import person.Student;
import person.consciousness.Knowledge;

public class Internship {
    private String internshipName;
    private int averageKnowledge;  
    public List<Student> internsList;

    public Internship(String name, int averageKnowledge) {
        this.internshipName = name;
        this.averageKnowledge = averageKnowledge;
        this.internsList = new ArrayList<Student>();
    }

    public void setStudent(Student student) {
		if(student.getKnowledge() > this.averageKnowledge){
            internsList.add(student);
        }
    }

    public String getStudents(){
    	String students = "";
        for (Student student : this.internsList) {
            students += student.getStudentName() + "\n";
        }
        return students;
    }
}

